titanic-dataset.ipynb

```diff
@@ -0,0 +1 @@
y6Pj+pWJyqFjJhgLVhqwGUDUxSoAsGbACefgELmRAlJgfhhsOAhkKsguh8x21/jxuHhXDwkDrfXj8XlJSlUT+i+SPL5g+I4MycYh9vPjb39JOS9yWUDDk6w1AoZmqhs9ox/nQfXyFlV6WVIYijXZHt5KnoV8sYCyZPF72PzBetYUafDPuRWkm1bucP/CRpcbNzbwLlZKiJ3vU/K8ju4q+kxZqTIDvjUAPSL0R4IxgD0jTMRu+czSvQ5uA0J1I9+nM9y3+O0dblMWBzHP5yXkRSm5x9n5GBqLZCCMUHxkDWdjCAfdW1OYoI0vDbVyPMR86mziZzeJ4YN5VZsbi8ymcDNw0K5JKaGlUWNKOUCf5ueRU2rkw9/LUEpl5Nf3cqWihY+XlfJRYMTeHN9C0O+MfJB0nPM7fMm1oSJZEZoUchlrPeldbgw2offDboTLQ7516TaZ8I/5Pp2baIplkZdUvcM6DB8lhpiFt/GlTsu5+vEL3lX9wrDFp6NomZjdw8NAK8osiCvpkP7hpKmbhjNX4vSRjtRpkOUUtVbT72h76FE5kDJymNuolXJGZRoZt7mymNuF+DkY/VARZ/DJvtKLTVBAzrtm1fn5LOWbO7TPsYD5hd4qXEwm+sF3L5j+8/0idZx44horv60kOlvbee/v1bwxRV9iDMfh+lo1Wb48Cx4ZSC8OxVKf+28T4A/JLWaFLxB7csINve+imoh8oT3pXU3I9/yYYd2RcmyLo2tqsVxIBiznx1VbdRZu5hG3FAECx+GtyfCO5NgyePQtLdr+wpwamjYDQOvZsmuOuxuH0/8kE+TzU3/eDOlyuQO82/fyLsI2vImqqp1kDoR+pyLzSdjTrGKsUsSeCD4BT5O+xflVqAu/2DHta9LAeij4Pb6+GB1SYf2+Ts6zskOJznMwN9Oy+bbW0bx7Hm59IppX5m1pMHGPV9sZfwLv3DTh+uwbP0W3hoHrwyAzy6Hul1H3nGAAL8jgdD0iaLSw+T/g+TRUPCTZPiXOR30YZiAMIOa+BAtm8qauHZMClkGJ1UuFaGuSu4fYSL6x4vB1sCNfY30nnga88sVnDUgibc2ViOXxzF/dy13T4nGZmtjasm/UWp1EDsIgHp5JJvqBXqp6zBYS2DN68gHzmZqaD2ldUrCjBpErRnBUoM8dSx9Fszik3PfY16xgN3rI0Pf3mnf7vbjDY9EHxLHv34u5pax07j/q3yUcoFzB8QRFaShxCLwzE/beKmPmkiAAZfDyhfpM9SBJ+ZMhp2TSf8N97E14XJqbH40Shl+UUQuCFQ02bkiroaQjS8RnPwmlwxN5LP15ZQ3OXj8rBw+WF1C/wQzQVolNW1O9tRb+ccZ2SjlMkJ1KmLNWiwN23hkIbx6mpm//yqiG/UBWY0L0dqrKI0/E3/IEAJ1TI6PIIUXQa6GsfdB5UYITkIwRhEitnX30ACQK9T4tWZkDbsJ2vnRgXZR2TNSghTIiNR3TJGJ1Afi2icTt9dPncV50NTXY5dW1bqzbGVwPLgt0jiO4WMzKl1KW5o9OlCFqzvRqFS8UZPMuRPeJbFiHi51GLsjTqPMEkZiJ31NGiVP/XjohLyVG8eloFYe2yi1trGZGz4vwOeXjBeWF7fhXVDMqxfoMIccYxHBUgufXSadWwCNhfC/WXDdMggNnEd/NgodQezq+xKDHSsJad5GZfQU5ramM1Hswn1FoUHUhXXU3Gi7tmgVouyYDiITIEhxfP4dHShaKD1472fVvyWlY3eYswc4PpQ6aCwkzDD1QNOW8ha2lLegF8wMGvcg1OwASxXEDcUdPRjt4kekDevyYeN7uM/9nAcXSwtXn7Y4YXsbNw4dzH3ByQgt+wJyujBJsliXB267dF7oD6piZYJApLFj+noHs/+jIJMJRzQP9vj8vL2imO+2VQNwW18Pxm+uOGhAvGcRLBBh1kcBpUyAk0rgSaIrBMXBwCvh4s9gxK0Qmnrgpd6xJs7rH8uSXfUMUJQyovQ1apvbyMp/GbOr+kAEOHLb65y37kLe9v6dOJ2bxDAD83dKBpBurx+9qwFtyWIo/BlCUiEkhbqIkZQ3OzD5mlEULwZAYYomon4NU3vHYJa7EXIvliT1Cg2FA//O35e2sqK4Ba1SSXrpp1zY23hgrHf386NOHILbYaHO4iK/TjLdundqFpvKmnl5SREbSpvJq25jnSsBnykBfC6IzoXd8xnw+VCGNn2LqmI1ZfJEksIkVdDuWgtapRy1Uo6mdAlUb2F8opoxGWGUNNi5b1om1a1OluyqJyPSSFq4Ab1Kzt4GO1anlxWF9aza08i81TupbLJQWGclr87O1f0MXPyjm5mF0ziv8XoeyYsmLLy950iAo2Nw1SKsfB5+fRnszVDwAyx9EmXLns47nwJaPHJK+t/Xrs1vjKVK33lp2lOB4PcwO9uLUn5wumtQK5gSdaRcugC/FxXNdsIMahT7zVC70z9mP4IMIntD8bFXnrOjTbQ6POyo7Bk+TX9VPC4Hw9MjmDFfw/kN13JW8Qxu+gVSwzv3wthR2dKhbUl+PSrvsUsJlzbaDwRj9vNriYW61k6qe7SUHQzG7MfVBs0BJcGfEZ1GwX/zFczYOozzWm9j+rI4HOpwZF0oe+1wexB6zZTKpu9Ha+6ymjBTXsVp6e0fQq8dGESGv/TEd+bzQv63HdsLfurS2AKcIsKzQKFicq8odKqD55VaIePcFD/8eC/sWQy2Bqzo0PzyePv+oh+xrbrDbr/Kt9KUdJr0iyDA+L/B+nfgjVGSguq96VCbd2B7hVzG7NHJ7eZfRrWCyb1OXEl2KA1WF99sOWjuHuurbF8NCqT313YcBvABAvwGAgqZLmBzeSmstdBkcxMfoiM13IBsX3lLq8vHGb1C0Pvb6L/tHuR+D5p0OU0RQ/CqTPsm8jmQfaYU3JCrMWqUNNpaSQrVMzw1lJRwPTtK6phsjAZLNSx9EiY9itYQxLq9xUwJUUBQLK7Y4fjix+C2NvLyikr+b1wor
```

Comment:
Nossa pensei que seria melhor de revisar um notebook aqui no github, mas fica como uma linha inteira kkkrying.
Então vamo la:
1. Existem comentários em todo o notebook que ultrapassam o limite de caracteres que daria pra ver no github e ai a pessoa que abriria o arquivo ve um comentário cortado e nem consegue rolar a tela pro lado. Vide exemplo abaixo
![image](https://github.com/pemariano/TitanicML/assets/61845573/bf48c050-6332-4fc4-be13-22f1a9210430)
No python alguns formatadores recomendam 80 caracteres por linha, mas não sei quanto é em um notebook teria que testar. O melhor é ir quebrando os comentários no meio.

2. Geralmente você mantém um projeto no github que alguem consiga baixar ele e reproduzir só com suas instruções ou arquivos. No notebook indica que existe uma pasta `/kaggle/...`, mas aqui no seu projeto não tem. Nos próximos projetos seria bom pensar nisso de reprodutibilidade.
3. A linha 3 `# ideas for this notebook:` poderia ser uma Header e não uma linha de código com comentários, notebook é bom pra isso, misturar textos e imagens com código, cada um no seu formato.
4. A título de curiosidade daria pra substituir a formatação da linha `pd.options.display.float_format = "{:.3f}".format # show only three decimals` de `"{:.3f}".format` para `f"{:.3f}"`. Não tenho 100% de certeza que assim rola, mas acho que sim.
5. Alguns comentários de `# analysis` ou `# questions` nas linhas 8 e 9 que podiam ser linhas de texto ao invés de código.

Reply:
Descobri que tem uma forma de fazer um review melhor de um notebook, tem tipo um plugin pra isso.

1. Geralmente comentários no código n é pra escrever textoes, é pra ser raro isso. Se precisar escrever um comentario que ultrapasse a linha era bom varias linhas com #. Vc pode colocar comentarios mais concisos no código e explicar com mais detalhes nas caixas de texto do notebook.

Reply:
Beleza, vou fazer isso das próximas.